pkg/upgrade/uninstallation.go

```diff
@@ -102,6 +109,16 @@ func removeDSCInitialization(ctx context.Context, cli client.Client) error {
 	return multiErr.ErrorOrNil()
 }
 
+func removeDSC(ctx context.Context, cli client.Client) error {
+	instance := &dscv1.DataScienceCluster{}
+
+	if err := cli.DeleteAllOf(ctx, instance, client.PropagationPolicy(metav1.DeletePropagationForeground)); err != nil {
```

Comment:
any reason to use  DeleteAllOf() here? I think in DSC controller, valid  number is checked on 1

Reply:
Yes, but this would enable us to skip the process of checking if there is any DSC controller and then deleting it. It just deletes all objects. https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client@v0.17.5#Writer.DeleteAllOf

Probably @lburgazzoli can add any other advantages if I've missed.

Reply:
right, i was wondering why it is a different way to handle deletion of DSC and DSCI differently. 
as in removeDSCInitialization() it still gets list and loop over (even in realiaty either it is 0 or 1). maybe that should be updated as well?

Reply:
i made https://github.com/opendatahub-io/opendatahub-operator/pull/1476 to follow a bit on the DSCI deletion part.